projects/driver_controls/test/test_ads1015.c

```diff
@@ -0,0 +1,97 @@
+#include <stdint.h>
+#include <stdio.h>
+#include "ads1015.h"
+#include "ads1015_def.h"
+#include "delay.h"
+#include "gpio_it.h"
+#include "i2c.h"
+#include "interrupt.h"
+#include "log.h"
+#include "soft_timer.h"
+#include "unity.h"
+
+static void prv_callback(Ads1015Channel channel, void *context) {
+  uint8_t *callback_called = context;
+  (*callback_called) = true;
+}
+
+void setup_test(void) {
+  gpio_init();
+  interrupt_init();
+  gpio_it_init();
+  I2CSettings i2c_settings = {
+    .speed = I2C_SPEED_FAST, .scl = { GPIO_PORT_B, 10 }, .sda = { GPIO_PORT_B, 11 }
+  };
+  i2c_init(I2C_PORT_2, &i2c_settings);
+}
+
+void teardown_test(void) {}
+
+void test_ads1015_init_invalid_input(void) {
+  GPIOAddress ready_pin = { GPIO_PORT_B, 2 };
+  Ads1015Storage storage;
+  TEST_ASSERT_EQUAL(STATUS_CODE_OK,
+                    ads1015_init(&storage, I2C_PORT_2, ADS1015_ADDRESS_GND, &ready_pin));
+  TEST_ASSERT_EQUAL(STATUS_CODE_INVALID_ARGS,
+                    ads1015_init(NULL, I2C_PORT_2, ADS1015_ADDRESS_GND, &ready_pin));
+  TEST_ASSERT_EQUAL(STATUS_CODE_INVALID_ARGS,
+                    ads1015_init(&storage, I2C_PORT_2, ADS1015_ADDRESS_GND, NULL));
+}
+
+void test_ads_config_channel_invalid_input(void) {
+  Ads1015Storage storage;
+  GPIOAddress ready_pin = { GPIO_PORT_B, 2 };
+
+  ads1015_init(&storage, I2C_PORT_2, ADS1015_ADDRESS_GND, &ready_pin);
+  TEST_ASSERT_EQUAL(STATUS_CODE_OK, ads1015_configure_channel(&storage, ADS1015_CHANNEL_0, true,
```

Comment:
Can you make sure your test name reflects what you're testing and add some brief comments explaining what each chunk is checking for?